cl/search/management/commands/sweep_indexer.py

```diff
@@ -0,0 +1,418 @@
+from datetime import datetime
+from typing import Iterable, Literal, Mapping, cast
+
+from django.apps import apps
+from django.conf import settings
+from django.db.models import QuerySet
+
+from cl.audio.models import Audio
+from cl.lib.celery_utils import CeleryThrottle
+from cl.lib.command_utils import VerboseCommand, logger
+from cl.lib.redis_utils import get_redis_interface
+from cl.people_db.models import Person
+from cl.search.documents import (
+    ES_CHILD_ID,
+    AudioDocument,
+    DocketDocument,
+    ESRECAPDocument,
+    OpinionClusterDocument,
+    OpinionDocument,
+    PersonDocument,
+)
+from cl.search.models import SEARCH_TYPES, Opinion, RECAPDocument
+from cl.search.tasks import (
+    index_parent_and_child_docs,
+    index_parent_or_child_docs,
+)
+from cl.search.types import ESDocumentClassType
+
+supported_models = [
+    "audio.Audio",
+    "people_db.Person",
+    "search.OpinionCluster",
+    "search.Opinion",
+    "search.Docket",
+    "search.RECAPDocument",
+]
+r = get_redis_interface("CACHE")
+
+
+def compose_indexer_redis_key() -> str:
+    """Compose the redis key for the sweep indexer
+    :return: A Redis key as a string.
+    """
+    return f"es_sweep_indexer:log"
+
+
+def log_indexer_last_status(
+    model_name: str, document_pk: int, chunk_size: int, log_key: str
+) -> Mapping[str | bytes, int | str]:
+    """Log the sweep indexer last status to Redis.
+
+    :param model_name: The document name being processed.
+    :param document_pk: The last document_id processed.
+    :param chunk_size: The last chunk size being processed.
+    :param log_key: The log key to use in redis.
+    :return: The data logged to redis.
+    """
+
+    stored_values = r.hgetall(log_key)
+    # Build the documents key containing the documents indexed dynamically.
+    documents_dict = {model: 0 for model in supported_models}
+    for model in supported_models:
+        current_total = int(stored_values.get(model, 0))
+        if model_name == model:
+            current_total = chunk_size + current_total
+        documents_dict[model] = current_total
+
+    data_to_log = {
+        "model_name": model_name,
+        "last_document_id": document_pk,
+        "date_time": datetime.now().isoformat(),
+    }
+    data_to_log.update(documents_dict)
+    log_info = cast(Mapping[str | bytes, int | str], data_to_log)
```

Comment:
The `data_to_log` variable seems to use only `string` keys. Are keys of type `bytes` possible in this scenario? Or is the mapping not updated?

Reply:
Yeah, the mapping was not updated. I just fixed it. Thanks for noticing it.